### ISSUE TITLE
Request: Mode to print error and suggestions

### ISSUE THREAD

**Issue by getaaron** [bbatsov/rubocop#2962]
I want to use RuboCop with Phabricator's ability to apply patches. That is, I want `arc lint`, which calls `rubocop`, to automatically fix linter errors when possible. The [related Phabricator issue is here](https://secure.phabricator.com/T10605). To do this I need to see the errors and replacement text at the same time.

Possibly related to https://github.com/bbatsov/rubocop/issues/2932.

## Expected behavior

I would like a mode that outputs original text and suggestions, without applying any actual changes.

```bash
$ rubocop --suggest-replacement-text something.rb
This text is bad: aaaa
Replace it with this: bbbb
```

From a comment on [the Phabricator ticket](https://secure.phabricator.com/T10605), regarding RuboCop's current functionality:

> It appears to support autocorrect, but not a mode where it prints the autocorrections without applying them. We can't deduce the corrections from a pre-corrected and post-corrected file because two or more correction ranges may overlap.

## Actual behavior

This feature appears to be unavailable (or undocumented). I infer from [the `RuboCop::Cop::Offense` docs](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Offense) that a custom "formatter" class can't output the correct information.

## Steps to reproduce the problem

N/A. This is a feature request.

## RuboCop version

```bash
$ rubocop -V
0.30.0 (using Parser 2.2.0.3, running on ruby 2.1.7 x86_64-darwin15.0)
```

**Comment by getaaron** [comment]
Seems like nobody's interested. Closing
